lib/ansible/modules/cloud/cloudscale/cloudscale_server.py

```diff
@@ -0,0 +1,438 @@
+
+    def __init__(self, module, api_token):
+        self._module = module
+        self._auth_header = {'Authorization': 'Bearer %s' % api_token}
+
+        # Check if server already exists and load properties
+        uuid = self._module.params['uuid']
+        name = self._module.params['name']
+
+        # Initialize server dictionary
+        self.info = {'uuid': uuid, 'name': name, 'state': 'absent'}
+
+        servers = self.list_servers()
+        matching_server = []
+        for s in servers:
+            if uuid:
+                # Look for server by UUID if given
+                if s['uuid'] == uuid:
+                    self.info =  self._transform_state(s)
+                    break
+            else:
+                # Look for server by name
+                if s['name'] == name:
+                    matching_server.append(s)
+        else:
+            if len(matching_server) == 1:
+                self.info = self._transform_state(matching_server[0])
+            elif len(matching_server) > 1:
+                self._module.fail_json(msg="More than one server with name '%s' exists. "
+                                       "Use the 'uuid' parameter to identify the server" % name)
+
+
+    def _get(self, api_call):
+        resp, info = fetch_url(self._module, API_URL+api_call, headers=self._auth_header)
+
+        if info['status'] == 200:
+            return json.loads(resp.read())
+        else:
+            self._module.fail_json(msg='Failure while calling the cloudscale.ch API with GET for '
+                                       '"%s": %s' % (api_call, info['body']))
+
+
+    def _post(self, api_call, data=None):
+        if data is not None:
+            data = urlencode(data)
+
+        resp, info = fetch_url(self._module,
+                               API_URL+api_call,
+                               headers = self._auth_header,
+                               method='POST',
+                               data=data)
+
+        if info['status'] == 201:
+            return json.loads(resp.read())
+        elif info['status'] == 204:
+            return None
+        else:
+            self._module.fail_json(msg='Failure while calling the cloudscale.ch API with POST for '
+                                       '"%s": %s' % (api_call, info['body']))
+
+
+    def _delete(self, api_call):
+        resp, info = fetch_url(self._module,
+                               API_URL+api_call,
+                               headers = self._auth_header,
+                               method='DELETE')
+
+        if info['status'] == 204:
+            return None
+        else:
+            self._module.fail_json(msg='Failure while calling the cloudscale.ch API with DELETE for '
+                                       '"%s": %s' % (api_call, info['body']))
+
+
+    @staticmethod
+    def _transform_state(server):
+        if 'status' in server:
+            server['state'] = server['status']
+            server.pop('status')
```

Comment:
because otherwise we would have state and status in the result. This is basically because our API calls the attribute status while the ansible convention is to call it state. So we want to rename the key in the dictionary.